### PR TITLE
fix: term vs trunc

### PR DESCRIPTION
**Pull request by sash-a**
## Note
This is quite confusing so please check this carefully and make sure I haven't mixed up where to use each one!

## What?
Mostly explained in #951. We are mixing up termination and truncation. This is especially prevalent in recurrent systems, but all systems have this issue.

Now very explicit about termination vs truncation, removed all mentions of done.
Done attribute remove from `RnnLearnerState` and replaced with `truncated` in the `PpoTransition`.

## How?
Explicitly name variables `termination` and `trunctation` instead of `done`

### CODE REVIEW

File: mava/advanced_usage/ff_ippo_store_experience.py [hunk 1]
```diff
@@ -147,39 +147,46 @@ def _env_step(learner_state: LearnerState, _: Any) -> Tuple[LearnerState, PPOTra
...
             env_state, timestep = jax.vmap(env.step, in_axes=(0, 0))(env_state, action)
 
-            # LOG EPISODE METRICS
-            done = jax.tree_util.tree_map(
-                lambda x: jnp.repeat(x, config.system.num_agents).reshape(config.arch.num_envs, -1),
-                timestep.last(),
-            )
+            trunc = jnp.repeat(timestep.last(), config.system.num_agents)
+            trunc = trunc.reshape(config.arch.num_envs, -1)
+            term = 1 - timestep.discount
```
Comment (OmaymaMahjoub) [inline mava/advanced_usage/ff_ippo_store_experience.py:162]: If you can add documentation to `timestep.last()` and `timestep.discount`

File: mava/systems/ff_ippo.py [hunk 2]
```diff
@@ -82,38 +82,45 @@ def _env_step(learner_state: LearnerState, _: Any) -> Tuple[LearnerState, PPOTra
...
+            # Log episode metrics
             info = {
                 "episode_return": env_state.episode_return_info,
                 "episode_length": env_state.episode_length_info,
             }
 
             transition = PPOTransition(
-                done, action, value, timestep.reward, log_prob, last_timestep.observation, info
+                terminal=term,
+                truncated=trunc,
```
Comment (OmaymaMahjoub) [inline mava/systems/ff_ippo.py:107]: Random question, why do we include truncation if we don't use it :sweat_smile:

File: mava/systems/rec_ippo.py
Comment (OmaymaMahjoub) [inline mava/systems/rec_ippo.py:1]: I really like the changes of removing `last_done` from the `learner_state`

File: mava/systems/rec_ippo.py [hunk 3]
```diff
@@ -90,19 +92,20 @@ def _env_step(
                 rng,
                 env_state,
                 last_timestep,
-                last_done,
                 hstates,
             ) = learner_state
 
             rng, policy_rng = jax.random.split(rng)
 
+            last_trunc = jnp.repeat(last_timestep.last(), n_agents).reshape(n_envs, -1)
```
Comment (OmaymaMahjoub) [inline mava/systems/rec_ippo.py:100]: if you can add documentation `# Add last_trunc to the input to the network.`

**Review by OmaymaMahjoub — COMMENTED** [review]
Thanks @sash-a! I will continue the review once we verify the recurrent systems performance :)

**Comment by sash-a** [comment]
This is an old attempt at fixing this. The most recent can be seen by following issue #951

**Pull request closed without merging**